File: langs/typescript/problems/rate_limiter/stub.ts
class Simulation {
  constructor() {}
  allow(key, timestamp) { throw new Error('not implemented'); }
  configure(key, limit, window) { throw new Error('not implemented'); }
  remaining(key, timestamp) { throw new Error('not implemented'); }
  allowWeighted(key, cost, timestamp) { throw new Error('not implemented'); }
}
module.exports = { Simulation };

File: langs/ruby/problems/rate_limiter/stub.rb
class Simulation
  def initialize
  end
  def allow(key, timestamp)
    raise 'not implemented'
  end
  def configure(key, limit, window)
    raise 'not implemented'
  end
  def remaining(key, timestamp)
    raise 'not implemented'
  end
  def allow_weighted(key, cost, timestamp)
    raise 'not implemented'
  end
end
